internal/infrastructure/kubernetes/proxy/resource_provider.go

```diff
@@ -80,6 +80,18 @@ func (r *ResourceRender) Service() (*corev1.Service, error) {
 				TargetPort: target,
 			}
 			ports = append(ports, p)
+
+			if port.Protocol == ir.HTTPSProtocolType {
```

Comment:
should this live here on in the Gateway API layer  ?

Reply:
This is for opening UDP ports for only those ports which have HTTPS configured, That's a requirement for quic. So this condition is added here.